### PR TITLE
Community Sidebar Events Links

### PR DESCRIPTION
**Pull request by lcummings12**
## Summary

Implemented a new partial, `primary-communities.html` to display the core COPs on any events page. Set the partial to display on community pages in the side bar. 

### Preview

[Link to Preview]()

### Solution
The `primary_communities`  partial filters and displays communities where the `dg_highlight` is set to `true`, signifying a "core" community. This partial is styled identitical to the `get_related` partial that this replaces. 

### How To Test
Proposed Change
Screen Shot 2024-06-28 at 1 13 57 PM

Production 
Screen Shot 2024-06-28 at 1 14 43 PM

1. Start a local server.
2. Open any Event page 
3. Assure the only difference between the production site and local site is the core COP sidebar

---

### Dev Checklist

- [x] PR has correct labels
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)
- [x] Code is formatted properly

### CODE REVIEW

File: themes/digital.gov/layouts/events/single.html [hunk 1]
```diff
@@ -146,6 +141,8 @@ <h1>{{- .Title | markdownify -}}</h1>
 
             <div class="dg-sidebar grid-col-12 tablet:grid-col-3">
               {{/* Authors */}}
+              {{- partial "core/primary_communities.html" . -}}
+
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/events/single.html:145]: This should come after the authors section. See live site — https://digital.gov/event/2022/05/17/how-to-build-your-digital-dream-team/

File: themes/digital.gov/layouts/partials/core/primary_communities.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{{ $mainCommunities := (where .Site.Pages "Section" "communities") }}
+{{ $mainCommunities = (where $mainCommunities ".Params.dg_highlight" "eq" true) }}
+{{ $featuredCommunity := (where $mainCommunities ".Params.weight" "eq" 2) }}
+{{ $mainCommunities = union $featuredCommunity $mainCommunities }}
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/partials/core/primary_communities.html:4]: Add comments at top of page explaining what this partial does. Comments will help explain the what and why for the code.

File: themes/digital.gov/layouts/partials/core/primary_communities.html
Comment (nick-mon1) [inline themes/digital.gov/layouts/partials/core/primary_communities.html:1]: Try renaming to `display_primary_communities.html` to be more descriptive.

**Review by nick-mon1 — CHANGES_REQUESTED** [review]

File: themes/digital.gov/layouts/partials/core/primary_communities.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/primary_communities.html:1]: ### Question

Where is this used? Unsure how this fits in with the previous communities in `get_related.html`.

https://github.com/GSA/digitalgov.gov/blob/a343de8cc9401a6e201a322f0349f3d82b1bcfba/themes/digital.gov/layouts/partials/core/get_related.html#L22-L59

**Review by mejiaj — COMMENTED** [review]
Thanks for helping us improve these components. Added a few comments below.

File: themes/digital.gov/layouts/partials/core/primary_communities.html
Comment (lcummings12) [reply themes/digital.gov/layouts/partials/core/primary_communities.html:1]: This was the part of the side bar that was present prior to these changes. At least in the events section, it is currently unused

Comment (mejiaj) [reply themes/digital.gov/layouts/partials/core/primary_communities.html:1]: Okay, thanks. We can look into whether it's worth combining or keeping separate in future work.

Comment (nick-mon1) [inline themes/digital.gov/layouts/partials/core/primary_communities.html:1]: ### Question

Should we remove this file? Looks to be same code as `primary-communities.html` minus the comments at the top.

**Review by nick-mon1 — APPROVED** [review]
Looks good, just have a question about removing the duplicate partial.

File: themes/digital.gov/layouts/partials/core/primary_communities.html
Comment (lcummings12) [reply themes/digital.gov/layouts/partials/core/primary_communities.html:1]: Oh yeah that carried over from my other branch. ill remove it now.

**Review by nick-mon1 — APPROVED** [review]

**Review by mejiaj — APPROVED** [review]
My comments have been resolved and visually looks good. I tested this [preview link](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Update-Community-Links-Events/event/2024/06/20/uswds-monthly-call-june-2024/).

---

@lcummings12 for future PR's can you add the Cloud Pages preview link in your PR description?

It gets added on every PR:
image

**Review by RileySeaburg — APPROVED** [review]
Great job addressing all the feedback. 

Thank you for helping us to improve this important component!

**Pull request merged**